.infrastructure/security/rbac.yml

```diff
@@ -0,0 +1,33 @@
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: secret-lister
+  namespace: todoapp
+
+---
+
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: secret-lister-role
+  namespace: todoapp
+rules:
+  - apiGroups: [""]
+    resources: ["secrets"]
+    verbs: ["list"]
+
+---
+
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: secret-lister-binding
+  namespace: todoapp
+subjects:
+  - kind: ServiceAccount
+    name: secret-lister
+    namespace: todoapp
+roleRef:
+  kind: Role
```

Comment:
The 'roleRef' field should not be a list. Remove the dash '-' to correctly define the role reference.